pyomo/gdp/tests/test_bigm.py

```diff
@@ -2193,6 +2248,46 @@ def test_decl_order_opposite_instantiation_order(self):
     def test_do_not_assume_nested_indicators_local(self):
         ct.check_do_not_assume_nested_indicators_local(self, 'gdp.bigm')
 
+    @unittest.skipUnless(gurobi_available, "Gurobi is not available")
+    def test_constraints_not_enforced_when_an_ancestor_indicator_is_False(self):
+        m = ConcreteModel()
+        m.x = Var(bounds=(0, 30))
+
+        m.left = Disjunct()
+        m.left.left = Disjunct()
+        m.left.left.c = Constraint(expr=m.x >= 10)
+        m.left.right = Disjunct()
+        m.left.right.c = Constraint(expr=m.x >= 9)
```

Comment:
I have a deep hatred of this. Not asking for it to change - it's just like directions from my nightmares.